### ISSUE TITLE
`library(<current package)` in package examples is obsolete

### ISSUE THREAD

**Issue by pawelru** [r-lib/lintr#2513]
Only after https://github.com/r-lib/lintr/issues/265

As an extra:
- `library()` calls to packages in Depends are also obsolete

**Comment by MichaelChirico** [comment]
Thanks for the issue! Could you clarify with a minimal example?

IIUC you'd like to throw a lint for

```r
#' @examples
#' library(PKG)
```

Within the source code for `PKG`, is that right?

**Comment by pawelru** [comment]
Yes that's correct. Sorry for not being crystal clear but you got it right.

**Comment by MichaelChirico** [comment]
Great! I marked as `new-linter`, but maybe it could fit under `namespace_linter()` too.
